internal/repository/sync_test.go

```diff
@@ -234,4 +234,56 @@ func TestClone(t *testing.T) {
 		}
 		assert.Equal(t, remotePolicyRef.Hash(), localPolicyRef.Hash())
 	})
+
+	t.Run("successful clone without specifying dir, with expected root keys", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
+
+		publicKey, err := tuf.LoadKeyFromBytes(rootKeyBytes)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		repo, err := Clone(context.Background(), remoteTmpDir, "", "", []*tuf.Key{publicKey})
+		assert.Nil(t, err)
+
+		head, err := repo.r.Head()
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, commitID, head.Hash())
+
+		localRSLRef, err := repo.r.Reference(plumbing.ReferenceName(rsl.Ref), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remoteRSLRef.Hash(), localRSLRef.Hash())
+
+		localPolicyRef, err := repo.r.Reference(plumbing.ReferenceName(policy.PolicyRef), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remotePolicyRef.Hash(), localPolicyRef.Hash())
+	})
+
+	t.Run("unsuccessful clone without specifying dir, with expected root keys not equaling root keys", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
+
+		expectedPublicKey, err := tuf.LoadKeyFromBytes(targetsKeyBytes)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = Clone(context.Background(), remoteTmpDir, "", "", []*tuf.Key{expectedPublicKey})
+		assert.Equal(t, err.Error(), ErrCloningRepository.Error()+", "+"cloned root keys do not match the expected keys.")
```

Comment:
Couple things:
a) technically, the first value in assert.Equal (after `t`) is the expected value, it's backwards here
b) we should be using assert.ErrorIs and also maybe having a named error for the keys not matching (returned using errors.Join) to avoid this concat here

Reply:
Fixed, I just named the whole joined error